src/components/ImageView/CatalogView/CatalogViewGLComponent.tsx

```diff
@@ -0,0 +1,362 @@
+            this.canvas.width = reqWidth;
+            this.canvas.height = reqHeight;
+            this.catalogWebGLService.setCanvasSize(reqWidth, reqHeight);
+        }
+        // Otherwise just clear it
+        this.gl.clearColor(0, 0, 0, 0);
+        const clearMask = WebGLRenderingContext.COLOR_BUFFER_BIT | WebGLRenderingContext.DEPTH_BUFFER_BIT | WebGLRenderingContext.STENCIL_BUFFER_BIT;
+        this.gl.clear(clearMask);
+    }
+
+    private updateCanvas = () => {
+        // const appStore = AppStore.Instance;
+        // const baseFrame = appStore.activeFrame;
+
+        if (this.canvas && this.gl && this.catalogWebGLService.shaderUniforms) {
+            this.resizeAndClearCanvas();
+            this.renderCatalog();
+            // draw in 2d canvas
+            const ctx = this.canvas.getContext("2d");
+            ctx.clearRect(0, 0, this.canvas.width, this.canvas.height);
+            ctx.drawImage(this.gl.canvas, 0, 0, this.canvas.width, this.canvas.height);
+        }
+    };
+
+    private renderCatalog() {
+        const appStore = AppStore.Instance;
+        const catalogStore = CatalogStore.Instance;
+        const GL2 = WebGL2RenderingContext;
+        // For alpha blending (soft lines)
+        this.gl.enable(GL2.BLEND);
+        this.gl.blendFunc(GL2.SRC_ALPHA, GL2.ONE_MINUS_SRC_ALPHA);
+        // depth test
+        this.gl.enable(GL2.DEPTH_TEST);
+        this.gl.depthFunc(GL2.LEQUAL);
+        this.gl.clear(GL2.COLOR_BUFFER_BIT | GL2.DEPTH_BUFFER_BIT);
+        const shaderUniforms = this.catalogWebGLService.shaderUniforms;
+        let rangeScale = {x: 1.0, y: 1.0};
+        let rangeOffset = {x: 0.0, y: 0.0};
+        let rotationAngle = 0.0;
+        let scaleAdjustment = 1.0;
+        const destinationFrame = appStore.activeFrame;
+        catalogStore.activeCatalogFiles.forEach(fileId => {
+            const frame = appStore.getFrame(catalogStore.getFramIdByCatalogId(fileId));
+            const isActive = frame === destinationFrame;
+            const catalog = catalogStore.catalogGLData.get(fileId);
+            if(catalog) {
+                const catalogWidgetStore = catalogStore.getCatalogWidgetStore(fileId);
+                const shape = catalogWidgetStore.shapeSettings;
+                const featherWidth = shape.featherWidth * devicePixelRatio;
+                const lineThickness = catalogWidgetStore.thickness * shape.thicknessBase * devicePixelRatio;
+                let color = tinycolor(catalogWidgetStore.catalogColor).toRgb();
+                let selectedSourceColor = tinycolor(catalogWidgetStore.highlightColor).toRgb();
+                let pointSize = catalogWidgetStore.catalogSize + shape.diameterBase;
+                this.gl.uniform1f(shaderUniforms.LineThickness, lineThickness);
+                this.gl.uniform1i(shaderUniforms.ShowSelectedSource, catalogWidgetStore.showSelectedData? 1.0 : 0.0);
+                //FrameView
+                let sourceFrame = frame;
```

Comment:
is sourceFrame actually appStore.activeFrame? If so, maybe we can just assign const sourceFrame = appStore.activeFrame;

Reply:
I think source frame is the frame that the catalog file is associated with. Not necessarily the same as the active frame.